ext/RMagick/rmpixel.c

```diff
@@ -172,6 +190,35 @@ DEF_PIXEL_CHANNEL_WRITER(green)
  */
 DEF_PIXEL_CHANNEL_WRITER(blue)
 
+/**
+ * Set Pixel alpha attribute.
+ *
+ * Ruby usage:
+ *   - @verbatim Pixel#alpha= @endverbatim
+ *
+ * Notes:
+ *   - Pixel is Observable. Setters call changed, notify_observers
+ *   - Setters return their argument values for backward compatibility to when
+ *     Pixel was a Struct class.
+ *
+ * @param self this object
+ * @param v the alpha value
+ * @return self
+ */
+VALUE
+Pixel_alpha_eq(VALUE self, VALUE v)
+{
+    Pixel *pixel;
+ 
+    rb_check_frozen(self);
+    Data_Get_Struct(self, Pixel, pixel);
```

Comment:
I'm seeing this pattern a lot in our C code and I'm wondering if it's a convention or if it's necessary in C. How come we don't have this method return a `Pixel` instead of modifying the pointer we send in? Would it be possible to do something like:

```c
pixel = Data_Get_Struct(self, Pixel);
```

or:

```c
Pixel *pixel = Data_Get_Struct(self, Pixel);
```

Just out of curiosity. I'm trying to get a better understanding of how these things work and why.

Reply:
We did not write that method `Data_Get_Struct` is from Ruby and that is why it has this signature.